### PR TITLE
ENG-6593: Introduce support for stream event cursors

### PR DESCRIPTION
**Pull request by erickpintor**
### Description

Provide support for the new per-event stream cursor.

### Motivation and context

There's an edge case where resuming a stream from a txn time with lots of documents written could result in event loss. This approach fixes that edge case.

### How was the change tested?

Unit tests included.

### Screenshots (if appropriate):

NA

### Change types

* - [X] Bug fix (non-breaking change that fixes an issue)
* - [X] New feature (non-breaking change that adds functionality)
* - [ ] Breaking change (backwards-incompatible fix or feature)

### Checklist:

* - [X] My code follows the code style of this project.
* - [X] My change requires a change to Fauna documentation.
* - [X] My change requires a change to the README, and I have updated it accordingly.

----
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -416,17 +416,27 @@ The [client configuration](#client-configuration) sets default query options for
 `Stream()`. To override these options, see [query
 options](#query-options).
 
-The `Subscribe()` method accepts a `fauna.StartTime` function. You can
-use `fauna.StartTime` to restart a stream after disconnection.
+The `Subscribe()` method accepts the `fauna.StartTime` and `fauna.EventCursor`
+function. You can use `fauna.StartTime` to restart a stream at a specific
+timestamp.
```
Comment (jrodewig) [inline README.md:421]:
```suggestion
function. Use `fauna.StartTime` to restart a stream at a specific
timestamp:
```

File: README.md [hunk 2]
```diff
@@ -416,17 +416,27 @@ The [client configuration](#client-configuration) sets default query options for
...
+```
+
 | Function | Description |
 | -------- | ----------- |
-| `fauna.StartTime`  | Sets the stream start time. Accepts an `int64` representing the start time in microseconds since the Unix epoch.<br><br>The start time is typically the time the stream disconnected.<br><br>The start time must be later than the creation time of the stream token. The period between the stream restart and the start time argument can't exceed the `history_days` value for source set's collection. If a collection's `history_days` is `0` or unset, the period can't exceed 15 minutes. |
+| `fauna.StartTime`  | Sets the stream start time. Accepts an `int64` representing the start time in microseconds since the Unix epoch.<br><br>The start time must be later than the creation time of the stream token. The period between the stream restart and the start time argument can't exceed the `history_days` value for source set's collection. If a collection's `history_days` is `0` or unset, the period can't exceed 15 minutes. |
+| `fauna.EventCursor`  | Resumes the stream after the given event cursor. Accepts an `string` with the cursor retrieved from a `fauna.Event`. |
```
Comment (jrodewig) [inline README.md:439]:
```suggestion
| `fauna.EventCursor`  | Resumes the stream after the given event cursor. Accepts a `string` with the cursor retrieved from a `fauna.Event`. |
```

File: README.md [hunk 3]
```diff
@@ -416,17 +416,27 @@ The [client configuration](#client-configuration) sets default query options for
...
+function. You can use `fauna.StartTime` to restart a stream at a specific
+timestamp.
 
 ```go
 streamQuery, _ := fauna.FQL(`Product.all().toStream()`, nil)
 client.Subscribe(streamQuery, fauna.StartTime(1710968002310000))
 ```
 
+The `fauna.EventCursor` function can be used to resume a stream after a
+disconnect.
```
Comment (jrodewig) [inline README.md:429]:
```suggestion
Use `fauna.EventCursor` to resume a stream after a disconnect:

**Review by jrodewig — APPROVED** [review]
Narrow 👍 . README looks good aside from a few nits.

File: README.md [hunk 4]
```diff
@@ -416,17 +416,25 @@ The [client configuration](#client-configuration) sets default query options for
...
+```
+
 | Function | Description |
 | -------- | ----------- |
-| `fauna.StartTime`  | Sets the stream start time. Accepts an `int64` representing the start time in microseconds since the Unix epoch.<br><br>The start time is typically the time the stream disconnected.<br><br>The start time must be later than the creation time of the stream token. The period between the stream restart and the start time argument can't exceed the `history_days` value for source set's collection. If a collection's `history_days` is `0` or unset, the period can't exceed 15 minutes. |
+| `fauna.StartTime`  | Sets the stream start time. Accepts an `int64` representing the start time in microseconds since the Unix epoch.<br><br>The start time must be later than the creation time of the stream token. The period between the stream restart and the start time argument can't exceed the `history_days` value for source set's collection. If a collection's `history_days` is `0` or unset, the period can't exceed 15 minutes. |
+| `fauna.EventCursor`  | Resumes the stream after the given event cursor. Accepts a `string` with the cursor retrieved from a `fauna.Event`. |
```
Comment (pnwpedro) [inline README.md:437]:
```suggestion
| `fauna.EventCursor`  | Resumes the stream after the given event cursor. Accepts a `string` representation of the cursor retrieved from a `fauna.Event`. |
```

File: config.go [hunk 5]
```diff
@@ -135,11 +135,19 @@ type StreamOptFn func(req *streamRequest)
...
-// Usefull when resuming a stream after a failure.
+// Usefull when resuming a stream at a given point in time.
 func StartTime(ts int64) StreamOptFn {
 	return func(req *streamRequest) { req.StartTS = ts }
 }
 
+// EventCursor set the stream starting point based on a previously received
+// event cursor.
+//
+// Usefull when resuming a stream after a failure.
```
Comment (pnwpedro) [inline config.go:146]:
```suggestion
// Useful when resuming a stream after a failure.
```

File: config.go [hunk 6]
```diff
@@ -135,11 +135,19 @@ type StreamOptFn func(req *streamRequest)
 
 // StartTime set the streams starting timestamp.
 //
-// Usefull when resuming a stream after a failure.
+// Usefull when resuming a stream at a given point in time.
```
Comment (pnwpedro) [inline config.go:138]:
```suggestion
// Useful when resuming a stream at a given point in time.
```

**Review by pnwpedro — CHANGES_REQUESTED** [review]
Only non-material feedback

**Comment by erickpintor** [comment]
I've addressed all feedback. This should be ready for another pass.
